### PR TITLE
[DllImportGenerator] Remove unnecessary DLLIMPORTGENERATOR_ENABLED ifdefs

### PR DESCRIPTION
**Pull request by elinor-fung**
Only went through the ones that are in projects that no longer even build down-level.
There's more we should be able to clean up now that we have the forwarding support.

cc @AaronRobinsonMSFT @jkoritzinsky

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Now that we have down level support, we should consider updating the code fix that uses this preprocessor option. I don’t know if it’s as useful as it was previously.

**Comment by elinor-fung** [comment]
Agreed - where updating means deleting that option?

I want to make sure we get through actually turning on / using the generator for all the down-level runtime libraries first though, and then do that as part of clean-up for productization.

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/libraries/Common/src/Interop/Windows/Advapi32/Interop.ConvertStringSidToSid.cs [hunk 1]
```diff
@@ -8,13 +8,8 @@ internal static partial class Interop
 {
     internal static partial class Advapi32
     {
-#if DLLIMPORTGENERATOR_ENABLED
-        [GeneratedDllImport(Interop.Libraries.Advapi32, EntryPoint = "ConvertStringSidToSidW", CharSet = CharSet.Unicode, SetLastError = true)]
+        [GeneratedDllImport(Interop.Libraries.Advapi32, EntryPoint = "ConvertStringSidToSidW", CharSet = CharSet.Unicode, ExactSpelling = true, SetLastError = true)]
```
Comment (stephentoub) [inline src/libraries/Common/src/Interop/Windows/Advapi32/Interop.ConvertStringSidToSid.cs:11]: Was the addition of ExactSpelling here just an opportunistic addition?  Should we be using it in other places we're not currently

File: src/libraries/Common/src/Interop/Windows/Advapi32/Interop.GetWindowsAccountDomainSid.cs [hunk 2]
```diff
@@ -8,13 +8,8 @@ internal static partial class Interop
 {
     internal static partial class Advapi32
     {
-#if DLLIMPORTGENERATOR_ENABLED
-        [GeneratedDllImport(Interop.Libraries.Advapi32, EntryPoint = "GetWindowsAccountDomainSid", CharSet = CharSet.Unicode, SetLastError = true)]
+        [GeneratedDllImport(Interop.Libraries.Advapi32, EntryPoint = "GetWindowsAccountDomainSid", SetLastError = true)]
         internal static partial int GetWindowsAccountDomainSid(
-#else
-        [DllImport(Interop.Libraries.Advapi32, EntryPoint = "GetWindowsAccountDomainSid", CharSet = CharSet.Unicode, SetLastError = true)]
-        internal static extern int GetWindowsAccountDomainSid(
```
Comment (stephentoub) [inline src/libraries/Common/src/Interop/Windows/Advapi32/Interop.GetWindowsAccountDomainSid.cs:16]: Do we have an analyzer that warns when specifying a CharSet and it's a nop? I assume that's how you spotted these? But if that's the case, why wasn't it previously firing for this #if case? Or is it just a suggestion right now that doesn't trigger warn-as-error build failures?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/Common/src/Interop/Windows/Advapi32/Interop.ConvertStringSidToSid.cs (hunk 1 above)
Comment (elinor-fung) [reply src/libraries/Common/src/Interop/Windows/Advapi32/Interop.ConvertStringSidToSid.cs:11]: Yeah, this was just opportunistic - I just noticed that already had the `W` suffix while doing this.

File: src/libraries/Common/src/Interop/Windows/Advapi32/Interop.GetWindowsAccountDomainSid.cs (hunk 2 above)
Comment (elinor-fung) [reply src/libraries/Common/src/Interop/Windows/Advapi32/Interop.GetWindowsAccountDomainSid.cs:16]: We don't. I just did a quick scan through the changes I had and manually updated a few things.

**Pull request merged**